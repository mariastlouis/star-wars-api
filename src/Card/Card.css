@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  background: black;
  color: white;
  font-family: "Cabin", sans-serif; }

h1 {
  font-size: 20px; }

h2 {
  display: block;
  -webkit-margin-before: 0;
  -webkit-margin-after: 0;
  -webkit-margin-start: 0px;
  -webkit-margin-end: 0px; }

.card-head {
  display: -ms-flexbox;
  display: flex;
  height: 20px;
  padding: 10px;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-right: 10px; }

.item-name h2 {
  margin: 0 20px;
  color: #feda4a;
  font-size: 20px;
  font-size: 20px; }

.item-name {
  display: flex;
  justify-content: center; }

.card {
  margin: 5px;
  border: 1px solid #feda4a;
  width: 230px;
  display: flex;
  flex-direction: column;
  background-color: #000024;
  font-size: 15px; }

.card-body {
  padding: 20px; }

ul.card-data {
  list-style-type: none;
  margin: 0 0 0 10px;
  padding: 0; }

li {
  font-size: 16px;
  color: gray; }

.favorite-star {
  width: 30px; }

.card-footer {
  display: flex;
  align-self: flex-end;
  justify-content: space-between; }

span.accent {
  color: white; }
