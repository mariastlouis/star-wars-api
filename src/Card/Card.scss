@import 'variable';

.card-head {
  display: flex;
  height: 75px;
  justify-content: space-between;
  text-align: center;
  padding-right: 10px;
}
.card-head h2 {
  font-size: 20px;
  margin: 0 20px;
}

.card {
  background: #ffffff;
  box-shadow: 1.5px 3px 3px 1.5px #566782;
  margin: 10px;
  width: 230px;
  color: black;
}

.card-body {
  padding: 20px;
}

ul.card-data {
  list-style-type: none;
  margin: 0 0 10px 35px;
  padding: 0;
}

li {
  color: black;
}

.accent {
  font-weight: 500%
}

.fav {
  height: 50px;
  width: 50px;
  // background-image: url("../images/fav-star-empty.png");
  // background-repeat: no-repeat;
}



