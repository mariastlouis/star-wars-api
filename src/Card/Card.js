import React from 'react';
import PropTypes from 'prop-types';
import './Card.scss';

const Card = ({name, data}) => {
  
const dataKeys = Object.keys(data);
const dataRows = dataKeys.map(key => {
 return <li> {`${key}: ${data[key]}`} </li> 

  });

  return (
    <div className = "card">
      <h2> {name} </h2>
      <div>
        <ul>
          {dataRows}
        </ul>
      </div>

      
    </div>
  )
}

export default Card;