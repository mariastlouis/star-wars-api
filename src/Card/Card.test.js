import React from 'react';
import Card from './Card.js';
import {shallow} from 'enzyme';

const mockAddFavorite = jest.fn();
const mockInfo= 
  {
   Species: 'Human',
    Language: 'Galactic Basic',
    Homeworld: 'Tatooine',
    Population: 200000
  }

  const mockInfoThree = {
    Class: 'wheeled',
    Model: 'Digger Crawler',
    Passengers: '30'
  }

  let renderedCard;

  describe('Card', () =>{
    beforeEach(() => {
      renderedCard = shallow(
        <Card 
          name = 'Luke Skywalker'
          info = {mockInfo}
          favorite = {false}
          addFavorite = {mockAddFavorite}
        />);
    });

    it('should render corrrectly', () => {
      expect(renderedCard).toBeDefined();
    });

  it('should have class of card head and fav an have an h2 element', () =>{
    expect(renderedCard.find('.card-head').length).toEqual(1);
    expect(renderedCard.find('h2').length).toEqual(1); 
    expect(renderedCard.find('.fav').length).toEqual(1);
  });

  it('should match the snapshot', () =>{
    expect(renderedCard).toMatchSnapshot()
  });

  it('should render an empty star image if the item is not a favorite', () =>{
    expect(renderedCard.containsMatchingElement(<img src="fav-star-empty.png" className="favorite-star" alt="not favorite" />)).toEqual(true);

  });
  it('should render a full star image if the item is a favorite', () =>{
    const renderedCard2 = shallow(
      <Card 
        name = 'Digger Crawler'
        info = {mockInfoThree}
        favorite = {true}
        addFavorite = {mockAddFavorite} /> );
 
    expect(renderedCard2.containsMatchingElement(<img src = "fav-star-full.png" className = "favorite-star" alt = "favorite" />)).toEqual(true);
   });

it ('should render the number of list items as there are pieces of info', () =>{
  expect(renderedCard.find('li').length).toEqual(4);
  const renderedCard2 = shallow(
      <Card 
        name = 'Digger Crawler'
        info = {mockInfoThree}
        favorite = {true}
        addFavorite = {mockAddFavorite} /> );
  expect(renderedCard2.find('li').length).toEqual(3);
})

it('should call the add favorite function if the favorite div is clicked', () =>{
  expect(mockAddFavorite.mock.calls.length).toEqual(0);
  renderedCard.find('.fav').simulate('click');
  expect(mockAddFavorite.mock.calls.length).toEqual(1);
})

});




