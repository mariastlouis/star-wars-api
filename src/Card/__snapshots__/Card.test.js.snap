// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    favorite={false}
    info={
        Object {
            "Homeworld": "Tatooine",
            "Language": "Galactic Basic",
            "Population": 200000,
            "Species": "Human",
          }
    }
    name="Luke Skywalker"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-head"
>
          <div
                    className="item-name"
          >
                    <h2>
                               
                              Luke Skywalker
                               
                    </h2>
          </div>
          <div
                    className="fav"
                    onClick={[Function]}
          >
                    <img
                              alt="not favorite"
                              className="favorite-star"
                              src="fav-star-empty.png"
                    />
          </div>
</div>,
        <div
          className="card-body"
>
          <ul
                    className="card-data"
          >
                    <li>
                               
                              <span
                                        className="accent"
                              >
                                        Species:
                              </span>
                               Human
                               
                    </li>
                    <li>
                               
                              <span
                                        className="accent"
                              >
                                        Language:
                              </span>
                               Galactic Basic
                               
                    </li>
                    <li>
                               
                              <span
                                        className="accent"
                              >
                                        Homeworld:
                              </span>
                               Tatooine
                               
                    </li>
                    <li>
                               
                              <span
                                        className="accent"
                              >
                                        Population:
                              </span>
                               200000
                               
                    </li>
          </ul>
</div>,
        <div
          className="card-footer"
/>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="item-name"
>
              <h2>
                             
                            Luke Skywalker
                             
              </h2>
</div>,
            <div
              className="fav"
              onClick={[Function]}
>
              <img
                            alt="not favorite"
                            className="favorite-star"
                            src="fav-star-empty.png"
              />
</div>,
          ],
          "className": "card-head",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2>
                 
                Luke Skywalker
                 
</h2>,
              "className": "item-name",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Luke Skywalker",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Luke Skywalker",
                " ",
              ],
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="not favorite"
                className="favorite-star"
                src="fav-star-empty.png"
/>,
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "not favorite",
                "className": "favorite-star",
                "src": "fav-star-empty.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="card-data"
>
            <li>
                         
                        <span
                                    className="accent"
                        >
                                    Species:
                        </span>
                         Human
                         
            </li>
            <li>
                         
                        <span
                                    className="accent"
                        >
                                    Language:
                        </span>
                         Galactic Basic
                         
            </li>
            <li>
                         
                        <span
                                    className="accent"
                        >
                                    Homeworld:
                        </span>
                         Tatooine
                         
            </li>
            <li>
                         
                        <span
                                    className="accent"
                        >
                                    Population:
                        </span>
                         200000
                         
            </li>
</ul>,
          "className": "card-body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                <span
                                className="accent"
                >
                                Species:
                </span>
                 Human
                 
</li>,
              <li>
                 
                <span
                                className="accent"
                >
                                Language:
                </span>
                 Galactic Basic
                 
</li>,
              <li>
                 
                <span
                                className="accent"
                >
                                Homeworld:
                </span>
                 Tatooine
                 
</li>,
              <li>
                 
                <span
                                className="accent"
                >
                                Population:
                </span>
                 200000
                 
</li>,
            ],
            "className": "card-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="accent"
>
                    Species:
</span>,
                  " Human",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Species:",
                    "className": "accent",
                  },
                  "ref": null,
                  "rendered": "Species:",
                  "type": "span",
                },
                " Human",
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="accent"
>
                    Language:
</span>,
                  " Galactic Basic",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Language:",
                    "className": "accent",
                  },
                  "ref": null,
                  "rendered": "Language:",
                  "type": "span",
                },
                " Galactic Basic",
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="accent"
>
                    Homeworld:
</span>,
                  " Tatooine",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Homeworld:",
                    "className": "accent",
                  },
                  "ref": null,
                  "rendered": "Homeworld:",
                  "type": "span",
                },
                " Tatooine",
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="accent"
>
                    Population:
</span>,
                  " 200000",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population:",
                    "className": "accent",
                  },
                  "ref": null,
                  "rendered": "Population:",
                  "type": "span",
                },
                " 200000",
                " ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "card-footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-head"
>
            <div
                        className="item-name"
            >
                        <h2>
                                     
                                    Luke Skywalker
                                     
                        </h2>
            </div>
            <div
                        className="fav"
                        onClick={[Function]}
            >
                        <img
                                    alt="not favorite"
                                    className="favorite-star"
                                    src="fav-star-empty.png"
                        />
            </div>
</div>,
          <div
            className="card-body"
>
            <ul
                        className="card-data"
            >
                        <li>
                                     
                                    <span
                                                className="accent"
                                    >
                                                Species:
                                    </span>
                                     Human
                                     
                        </li>
                        <li>
                                     
                                    <span
                                                className="accent"
                                    >
                                                Language:
                                    </span>
                                     Galactic Basic
                                     
                        </li>
                        <li>
                                     
                                    <span
                                                className="accent"
                                    >
                                                Homeworld:
                                    </span>
                                     Tatooine
                                     
                        </li>
                        <li>
                                     
                                    <span
                                                className="accent"
                                    >
                                                Population:
                                    </span>
                                     200000
                                     
                        </li>
            </ul>
</div>,
          <div
            className="card-footer"
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="item-name"
>
                <h2>
                                 
                                Luke Skywalker
                                 
                </h2>
</div>,
              <div
                className="fav"
                onClick={[Function]}
>
                <img
                                alt="not favorite"
                                className="favorite-star"
                                src="fav-star-empty.png"
                />
</div>,
            ],
            "className": "card-head",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                   
                  Luke Skywalker
                   
</h2>,
                "className": "item-name",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Luke Skywalker",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Luke Skywalker",
                  " ",
                ],
                "type": "h2",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="not favorite"
                  className="favorite-star"
                  src="fav-star-empty.png"
/>,
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "not favorite",
                  "className": "favorite-star",
                  "src": "fav-star-empty.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="card-data"
>
              <li>
                             
                            <span
                                          className="accent"
                            >
                                          Species:
                            </span>
                             Human
                             
              </li>
              <li>
                             
                            <span
                                          className="accent"
                            >
                                          Language:
                            </span>
                             Galactic Basic
                             
              </li>
              <li>
                             
                            <span
                                          className="accent"
                            >
                                          Homeworld:
                            </span>
                             Tatooine
                             
              </li>
              <li>
                             
                            <span
                                          className="accent"
                            >
                                          Population:
                            </span>
                             200000
                             
              </li>
</ul>,
            "className": "card-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                   
                  <span
                                    className="accent"
                  >
                                    Species:
                  </span>
                   Human
                   
</li>,
                <li>
                   
                  <span
                                    className="accent"
                  >
                                    Language:
                  </span>
                   Galactic Basic
                   
</li>,
                <li>
                   
                  <span
                                    className="accent"
                  >
                                    Homeworld:
                  </span>
                   Tatooine
                   
</li>,
                <li>
                   
                  <span
                                    className="accent"
                  >
                                    Population:
                  </span>
                   200000
                   
</li>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="accent"
>
                      Species:
</span>,
                    " Human",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Species:",
                      "className": "accent",
                    },
                    "ref": null,
                    "rendered": "Species:",
                    "type": "span",
                  },
                  " Human",
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="accent"
>
                      Language:
</span>,
                    " Galactic Basic",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Language:",
                      "className": "accent",
                    },
                    "ref": null,
                    "rendered": "Language:",
                    "type": "span",
                  },
                  " Galactic Basic",
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="accent"
>
                      Homeworld:
</span>,
                    " Tatooine",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Homeworld:",
                      "className": "accent",
                    },
                    "ref": null,
                    "rendered": "Homeworld:",
                    "type": "span",
                  },
                  " Tatooine",
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="accent"
>
                      Population:
</span>,
                    " 200000",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population:",
                      "className": "accent",
                    },
                    "ref": null,
                    "rendered": "Population:",
                    "type": "span",
                  },
                  " 200000",
                  " ",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "card-footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
